src/context/auth.tsx

```diff
@@ -0,0 +1,23 @@
+import { createContext, FC, useState } from 'react'
+import { IAuthContext, IAuthContextState } from '../interfaces/IAuth'
+
+export const AuthContext = createContext<IAuthContext | undefined>(undefined)
+
+const DEFAULT_USER = {
+  id: 0,
+  name: '',
+  email: '',
+  token: '',
+}
```

Comment:
not sure if this is good
maybe it's better to be null

Reply:
I got this feeling as well. 

I am concerned to pass something optional in my context and when I going to use that context I need to use a lot of optional chaining

```ts
const { authenticatedUser } = useAuthenticatedUser()

authenticatedUser?.name
```